game/scripts/vscripts/TutorialGraph/Steps.ts

```diff
@@ -468,3 +468,92 @@ export const textDialog = (text: tg.StepArgument<string>, unit: tg.StepArgument<
 export const neverComplete = () => {
     return tg.step(_ => { })
 }
+
+/**
+ * Creates a particle system at a location and optionally destroys it and complets after a given time. If no duration is passed it will never complete.
+ * @param particleName Name of the particle system.
+ * @param location Location to spawn the particles at.
+ * @param duration Optional duration after which to destroy the particles and complete.
+ */
+export const createParticleAtLocation = (particleName: tg.StepArgument<string>, location: tg.StepArgument<Vector>, duration?: tg.StepArgument<number>) => {
+    let timer: string | undefined = undefined
+    let particle: ParticleID | undefined = undefined
+
+    return tg.step((context, complete) => {
```

Comment:
Why do we need this to be a step? Couldn't we just make this a utility function that returns the particle ID? That way we can just call this from a tg.immediate and control the lifetime from the graph.

Reply:
Yes we could do that too. However this one also takes care of cleanup when the graph gets stopped which would need to be manually done then so I think it makes some sense having a step for it (which uses that util function perhaps).